scipy/linalg/_special_matrices.py

```diff
@@ -1396,3 +1398,339 @@ def convolution_matrix(a, n, mode='full'):
         col0 = az
         row0 = raz[-n:]
     return toeplitz(col0, row0)
+
+
+def laplacian_Nd(
+    shape, bc="N", form="sparse", return_evals=False, dtype="float"
+):
+    """
+    Construct a sparse matrix and a callable associated with the Laplacian
+    on a uniform rectangular grid in `N` dimensions given by tuple `shape`
+    and output its eigenvalues.
+
+    The Laplacian matrix `L` is square real symmetric negative definite.
+
+    Parameters
+    ----------
+    shape : tuple of int
+        The shape of the grid for the `N` dimensional Laplacian of length `N`
+        where the shape `n` of the Laplacian matrix `L` is ``np.prod(shape)``
+    bc: ``'D'`` or ``'N'`` or ``'P'``
+        The type of the boundary conditions on the boundaries of the grid
+    form: 'array', or 'sparse', or 'function', or 'lo'
+        Determines the format of the output Laplacian:
+
+        * 'array' is a numpy array;
+        * 'sparse' is a scipy sparse matrix;
+        * 'function' is a pointer to evaluating the Laplacian-vector
+          or Laplacian-matrix product;
+        * 'lo' results in the format of the `LinearOperator`.
+
+        Default: 'sparse'.
+    return_evals: logical 'True' or 'False'
+        If ``return_evals == 'True'`` also return all eigenvalues
```

Comment:
eval is IMO a very misleading abbreviation for eigenvalues. But more importantly, is this really necessary? Having variable signatures based on keywords is an anti-pattern that we're trying to get rid of, so we definitely shouldn't be adding more.

Reply:
Gone